onnxruntime/src/runner.rs

```diff
@@ -0,0 +1,244 @@
+use std::ffi::CString;
+use std::fmt::Debug;
+use std::ops::{Index, IndexMut};
+
+use ndarray::{Array, Dimension, IxDyn};
+
+use onnxruntime_sys as sys;
+
+use crate::error::{status_to_result, OrtError};
+use crate::memory::MemoryInfo;
+use crate::session::{Output, Session};
+use crate::tensor::OrtTensor;
+use crate::{g_ort, Result, TypeToTensorElementDataType};
+
+pub trait Element: 'static + Clone + Debug + TypeToTensorElementDataType + Default {}
+
+impl<T: 'static + Clone + Debug + TypeToTensorElementDataType + Default> Element for T {}
+
+fn names_to_ptrs(names: impl Iterator<Item = String>) -> Vec<*const i8> {
+    names
+        .map(|name| CString::new(name.clone()).unwrap().into_raw() as *const _)
+        .collect()
+}
+
+fn compute_output_shapes<TIn, DIn: Dimension>(
```

Comment:
I'd prefer if the trait bounds were using _where clauses_ to be consistent with the rest of the crates.